python/sglang/srt/lora/lora_manager.py

```diff
@@ -422,6 +422,28 @@ def init_lora_shapes(
 
         if target_modules is not None:
             self.target_modules = set(target_modules)
+            user_normalized_modules = get_normalized_target_modules(self.target_modules)
+            for lora_id, config in self.configs.items():
+                if not isinstance(config.target_modules, list):
+                    raise ValueError(
+                        f"SGLang currently only supports inferring LoRA target modules when a list of "
+                        "suffixes is provided in `target_modules` field of PEFT config. Please explicitly "
+                        "specify `--lora-target-modules` during server startup. You can specify `all` to "
+                        "enable all support modules types. "
+                    )
+                config_normalized_modules = get_normalized_target_modules(
+                    config.target_modules
+                )
+                if not config_normalized_modules.issubset(user_normalized_modules):
+                    unsupported_modules = (
+                        config_normalized_modules - user_normalized_modules
+                    )
+                    raise ValueError(
+                        f"LoRA adapter '{lora_id}' contains target modules {sorted(unsupported_modules)} "
+                        f"that are not included in the specified --lora-target-modules {sorted(user_normalized_modules)}. "
```

Comment:
You can get user-friendly lora_name in `self.lora_refs[id].lora_name`. lora_id is an auto-generated random UUID.